sdks/python/apache_beam/dataframe/frames_test.py

```diff
@@ -1245,6 +1245,37 @@ def test_idxmax(self):
     self._run_test(lambda s2: s2.idxmax(), s2)
     self._run_test(lambda s2: s2.idxmax(skipna=False), s2)
 
+  def test_pipe(self):
+    def df_times(df, column, times):
+      df[column] = df[column] * times
+      return df
+
+    def s_times(s, times):
+      return s * times
+
+    df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]}, index=[0, 1, 2])
+    s = pd.Series([1, 2, 3, 4, 5], index=[0, 1, 2, 3, 4])
+    # s2 = pd.Series([6, 7], index=[5, 6])
+
+    func_1 = df_times
+    func_2 = frames.DeferredDataFrame.sum
+    func_3 = frames.DeferredSeries.sum
```

Comment:
I'm very surprised that these work. `_run_test` works by first executing the lambda with normal pandas to compute the expected the result. I'd think that would fail in the pandas case, since it's trying to call the Beam function.

I think it's just a fluke that they work, since `sum` is implemented with `_agg_method`: https://github.com/apache/beam/blob/5ccb00298dcc591e0af7061a5ab0fbdb1196ca8c/sdks/python/apache_beam/dataframe/frames.py#L1957

Which defers to `self.agg`:
https://github.com/apache/beam/blob/5ccb00298dcc591e0af7061a5ab0fbdb1196ca8c/sdks/python/apache_beam/dataframe/frames.py#L131

The other tests you have here (`func_1`, `func_4`) are sufficient in my opinion, could you just remove the `frames.Deferred*` ones, since they're a little circular?

Reply:
Got it, I deleted all tests that had a `frames.Deferred` function as a parameter.